addons/hr_recruitment_skills/models/hr_applicant.py

```diff
@@ -1,10 +1,54 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
 
 
 class HrApplicant(models.Model):
     _inherit = 'hr.applicant'
 
     candidate_skill_ids = fields.One2many(related="candidate_id.candidate_skill_ids", readonly=False)
     skill_ids = fields.Many2many(related="candidate_id.skill_ids", readonly=False)
+    matching_score = fields.Float(string="Matching Score(%)", compute="_compute_matching_score", search="_search_matching_score",
+        help="The Matching Score reflects the match between the job positions's skills and those found in the applicant's resume.")
+    expected_skill_ids = fields.Many2many(related="job_id.skill_ids")
+
+    @api.depends('skill_ids')
+    def _compute_matching_score(self):
+        if not self.job_id.skill_ids:
+            self.matching_score = 0
+        else:
+            for applicant in self:
+                matching_skill_ids = applicant.job_id.skill_ids & applicant.skill_ids
+                applicant.matching_score = (len(matching_skill_ids) / len(applicant.job_id.skill_ids)) * 100
```

Comment:
if your goal is to show percentage then you can use `widget="percentage"` and remove `* 100`

Reply:
Hello @kais-odoo 
I've applied the changes :innocent: